src/widgets/modals/SearchForm.js

```diff
@@ -0,0 +1,159 @@
+            {type === 'date' ? value.toDateString() : value}
+        </Text>
+    </View>
+);
+
+const SearchListItem = ({ listItem, listConfig }) => {
+    const onPress = item => console.log(item);
+    const columns = listConfig.map(({ key, type}) => {
+        const value = listItem[key];
+        return <SearchListItemColumn key={key} value={value} type={type}/>;
+    });
+    return (
+        <TouchableOpacity onPress={onPress} style={localStyles.rowContainer}>
+            {columns}
+        </TouchableOpacity>
+    );
+}
+
+export const SearchForm = ({dataSource}) => {
+    const [data, setData] = useState([]);
+
+    if (!RECORD_TYPES.includes(dataSource)) return null;
+
+    const configKey = useMemo(() => FORM_CONFIGS[dataSource], [dataSource]);
+    const formConfig = useMemo(() => getFormInputConfig(configKey), [configKey]);
+
+    const listConfig = useMemo(() => {
+        const keys = LIST_CONFIGS[dataSource];
+        const keyTypes = keys.reduce((acc, key) => ({ ...acc, [key]: formConfig.find(config => config.key === key)?.type }), {});
+        return keys.map(key => ({ key, type: keyTypes[key] }));
+    }, [formConfig]);
+
+    const renderItem = useMemo(() => ({item}) => <SearchListItem key={item.id} listItem={item} listConfig={listConfig}/>, [listConfig]);
+
+    const lookupRecords = useMemo(() => params => {
+        switch (dataSource) {
+            case RECORD_TYPES.PATIENT: {
+                queryPatientApi(params).then(patientData => setData(patientData));
+                break;
+            }
+            case RECORD_TYPES.PRESCRIBER: {
+                queryPrescriberApi(params).then(prescriberData => setData(prescriberData));
+                break;
+            }
+            default: queryPatientApi(params);
+        }
+    }, []);
+
+    return (
+        <View style={localStyles.container}>
+            <View style={localStyles.formContainer}>
+                <FormControl
+                    isDisabled={false}
+                    isSearchForm={true}
+                    onSave={lookupRecords}
+                    onCancel={() => null}
+                    inputConfig={formConfig}
+                />
+            </View>
+            <View style={localStyles.verticalSeparator}></View>
+            <View style={localStyles.listContainer}>
+                <FlatList
+                    data={data}
+                    keyExtractor={record => record.id}
+                    renderItem={renderItem}
+                />
+            </View>
+        </View>
+    );
+  };
+  
+SearchForm.propTypes = {
+    dataSource: PropTypes.string.isRequired,
+};
+
+SearchListItem.propTypes = {
+    listItem: PropTypes.object.isRequired,
+    listConfig: PropTypes.array.isRequired,
+}
+
+SearchListItemColumn.propTypes = {
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.instanceOf(Date)]).isRequired,
+    type: PropTypes.string.isRequired,
+}
+
+  
+const localStyles = StyleSheet.create({
```

Comment:
Don't like the reliance on absolute values for the styles here, but gets the job done for now. Planning to address some of these in a dedicated UI tidy up issue once the main functionality is implemented.